pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -752,6 +752,14 @@ public CompletableFuture<Void> closeAsync() {
         final CompletableFuture<Void> closeFuture = new CompletableFuture<>();
         List<CompletableFuture<Void>> futures = new ArrayList<>();
 
+        urlLookupMap.entrySet().forEach(e -> {
+            try {
+                e.getValue().close();
+            } catch (Exception ex) {
+                log.error("Error closing lookup service {}", e.getKey(), ex);
+            }
+        });
+
```

Comment:
It would be worth creating a separate PR for this fix. The benefit is that it could be cherry-picked to older maintenance branches which won't be getting this current PR.

Reply:
Created a separate PR for this fix
- https://github.com/apache/pulsar/pull/22858